### PR TITLE
GH-39651: [Python] Basic pyarrow bindings for Binary/StringView classes

### LINKED ISSUE #39651: [Python] Binary/StringView: add basic support to represent the Array, DataType and Scalar classes

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
### Rationale for this change

First step for https://github.com/apache/arrow/issues/39633: exposing the Array, DataType and Scalar classes for BinaryView and StringView, such that those can already be represented in pyarrow.

(I exposed a variant of StringBuilder as well, just for now to be able to create test data)

* Closes: #39651

### CODE REVIEW

File: python/pyarrow/tests/test_builder.py [hunk 1]
```diff
@@ -65,3 +65,22 @@ def test_string_builder_append_after_finish():
     sbuilder.append("No effect")
     expected = [None, None, "text", None, "other text"]
     assert arr.to_pylist() == expected
+
```
[the remark this answers is not in the document]
Comment (jorisvandenbossche) [reply python/pyarrow/tests/test_builder.py:68]: I didn't add a binary builder (there is also no binary builder for the existing simple binary type). A next PR will add proper creation in the Python->Arrow conversion path, so I don't want to spend much time here on having a fully featured builder.

**Review by rok — APPROVED** [review]

File: python/pyarrow/array.pxi [hunk 2]
```diff
@@ -2943,6 +2943,12 @@ cdef class LargeStringArray(Array):
                                   null_count, offset)
 
 
+cdef class StringViewArray(Array):
+    """
+    Concrete class for Arrow arrays of string (or utf8) view data type.
+    """
```
Comment (danepitkin) [inline python/pyarrow/array.pxi:2949]: Is it worth adding TODOs or linking future GH issues?

**Review by danepitkin — COMMENTED** [review]
Overall LGTM!

File: python/pyarrow/array.pxi (hunk 2 above)
Comment (jorisvandenbossche) [reply python/pyarrow/array.pxi:2949]: It's unclear if we need to add any specific method here (at least it's not needed for the current TODO items in the parent issue), so going to leave this as is for now.

**Comment by kou** [comment]
@jorisvandenbossche It seems that AppVeyor started failing by this commit. Could you take a look at this?

**Comment by kou** [comment]
GH-39884

**Comment by jorisvandenbossche** [comment]
Ah I assumed it was an unrelated failure .. Indeed seems to have started with this PR, but not directly any idea what could cause this

**Pull request merged**